File: src/main/java/com/ject/studytrip/mission/application/dto/MissionInfo.java
package com.ject.studytrip.mission.application.dto;

import com.ject.studytrip.global.util.DateUtil;
import com.ject.studytrip.mission.domain.model.Mission;

public record MissionInfo(
        Long missionId,
        String missionName,
        boolean completed,
        String createdAt,
        String updatedAt,
        String deletedAt) {
    public static MissionInfo from(Mission mission) {
        return new MissionInfo(
                mission.getId(),
                mission.getName(),
                mission.isCompleted(),
                DateUtil.formatDateTime(mission.getCreatedAt()),
                DateUtil.formatDateTime(mission.getUpdatedAt()),
                DateUtil.formatDateTime(mission.getDeletedAt()));
    }
}

File: src/main/kotlin/com/ject/studytrip/mission/application/dto/MissionInfo.kt
package com.ject.studytrip.mission.application.dto

import com.ject.studytrip.global.util.DateUtil
import com.ject.studytrip.global.util.EntityExtensions.requireId
import com.ject.studytrip.mission.domain.model.Mission

data class MissionInfo(
    val missionId: Long,
    val missionName: String,
    val completed: Boolean,
    val createdAt: String,
    val updatedAt: String,
    val deletedAt: String?,
) {
    companion object {
        fun from(mission: Mission): MissionInfo =
            MissionInfo(
                mission.id.requireId(),
                mission.name,
                mission.isCompleted(),
                DateUtil.formatDateTime(requireNotNull(mission.createdAt)),
                DateUtil.formatDateTime(requireNotNull(mission.updatedAt)),
                mission.deletedAt?.let { DateUtil.formatDateTime(it) },
            )
    }
}
